### DISCUSSION TITLE
ConnectionFactory returns "None of the endpoints were reachable" only in one server

### DISCUSSION THREAD

**Discussion by JLopeDeB** [rabbitmq/rabbitmq-dotnet-client#1627] [Q&A]
Hello, I wonder if you could give me a hand on solved this issue. First, our env:

**RabbitMQ and Erlang version** RabbitMQ 3.13.3 Erlang 26.2.5
**Operating system for RabbitMQ and your client application.** RabbitMQ: Docker Container with tag 3-management; Client: Windows Server 2016 fully patched.
**RabbitMQ.Client library version** 6.8.1
**.NET version** 4.7.02053 (.NET Framework)
**Application runtime (Windows Server, IIS, etc):** Windows Server 2016, IIS

**FULL stack traces** // Not exactly sure what you mean, but here's a portion of the log from .net app:
```
Fecha: 03/07/2024 08:04:19 a. m.
Usuario: Sin Usuario ()
Grupo Usuario: Sin Grupo
Browser: Chrme 126.0
Sistema Opetaivo: WinNT

Direccion IP: 10.22.61.23
Cookies: True
IsSessionTimedOut: True

Tipo de Error: System.Exception
Descripcion: None of the specified endpoints were reachable
Origen: 
Peticion: http://myapp.mydomian.gob.ar/mycontroller.aspx
Tamaño ViewState: 432
Parametros: 
Pila: 
````
RabbitMQ Log;
[_rabbitmq_logs.txt](https://github.com/user-attachments/files/16097718/_rabbitmq_logs.txt)

### Problem
We have set up RabbitMQ for Prod and Dev environments, as Docker containers. Our app works fine, and is able to connect to the service from everywhere but one server with same OS and client app. We went to the entire troubleshooting [connectivity guide](https://www.rabbitmq.com/docs/troubleshooting-networking), and everything is fine. Ping is fine, telnet to 5672 is fine, even ncap works (you can see that in RabbitMQ log, attached, but is the only thing logged from the server). The only thing that doesn't work from that particular server, is connecting from our app to the RabbitMQ service ("None of the endpoints were reachable"). I believe this must be a combination of IIS config + Windows components, other than that there is no logical explanation I could find. The code is also working in a couple of servers / PCs, no issues there. 

There is a similar issue from couple of years back, [now closed](https://github.com/rabbitmq/rabbitmq-dotnet-client/discussions/1055).

Let me know if you need further details. Any help would be appreciated.

**Comment by lukebakken** [comment]
> Our app works fine, and is able to connect to the service from everywhere but one server with same OS and client app

It's an obvious question, but what is different about this "one server"?

> RabbitMQ: Docker Container with tag 3-management;

How are you running this docker container? On Windows, Linux, etc? I'm assuming it's on a different server than those hosting your client applications.

**Reply by JLopeDeB** [reply]
Thanks for your answer @lukebakken . 

> > Our app works fine, and is able to connect to the service from everywhere but one server with same OS and client app
> 
> It's an obvious question, but what is different about this "one server"?

That we couldn't figure out. At first I thought it was Windows related (2016 vs 2019 -that is now discarded-), but after several tests now I thinks is IIS / App pool related, although we weren't able to find evidence. Config on similar servers looks the same.

> 
> > RabbitMQ: Docker Container with tag 3-management;
> 
> How are you running this docker container? On Windows, Linux, etc? I'm assuming it's on a different server than those hosting your client applications.

Linux Docker container running on Ubuntu Server 22.04. So yes, it's a different server.

**Reply by lukebakken** [reply]
On the affected server, have you disabled all of the following:
* Anti-virus, and Windows Defender
* Windows Firewall

I would also be _absolutely certain_ that the application on this server is using the same configuration as the other servers. I'm just guessing, but since it is IIS I'm assuming you're using something like `web.config` to store RabbitMQ connection parameters. Perhaps the wrong configuration file is being read on this one server.

**Reply by JLopeDeB** [reply]
We tried disabling AV and Firewall (which by the way is the same on other servers) with no luck. 

As for the web.config / app config, we are certain is the same on other clients. Trust me we checked this serveral times. We don't believe the issue is RabbitMQ server, and most likely not the client DLL either, but was hoping someone run into similar issues with IIS.

Thanks again for your help.

**Reply by lukebakken** [reply]
> As for the web.config / app config, we are certain is the same on other clients. Trust me we checked this serveral times

Sure, but I bring it up because I vaguely remember an IIS/.NET "feature" where you can have multiple copies of those files and they will be merged into a final config. I may be imagining things.

Have you run Wireshark or similar traffic capture on this server to see if, in fact, connection attempts are being made?

**Reply by JLopeDeB** [reply]
Not Wireshark but other tools like our corp Firewall and the analysis showed no connection attempts to the server.

**Reply by lukebakken** [reply]
Well, without a way to reproduce, I'm out of ideas 🤷‍♂️ 

Please follow-up if you do figure this issue out!

**Reply by JLopeDeB** [reply]
Sure, thanks a lot for your help. I'll make sure to update the discussion in case we find the root cause.
